examples/machine_learning/geneticalgorithm.cpp

```diff
@@ -0,0 +1,181 @@
+/*******************************************************
+ * Copyright (c) 2014, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#include <arrayfire.h>
+#include <cstdio>
+
+using namespace af;
+
+array update(const array& searchSpace, const array& samplex, const array& sampley, const int nsamples, const int n)
+{
+    array result(nsamples, 1, f32);
+    result = searchSpace(sampley*n + samplex);
+    return result;
+}
+
+array selectFittest(const array& samplez, const int nsamples) {
+    //pick top 50% fittest
+    array sorted = sort(samplez);
+    array indices = where(samplez >= sorted.row(nsamples / 2).scalar<float>());
+
+    if (indices.elements() > nsamples / 2) {
+        indices = indices.rows(indices.elements() - nsamples / 2, indices.elements() - 1);
+    }
+
+    return indices;
```

Comment:
You can always pick to top 50% without doing where. If there are a bunch of variables around the midway point that have the same value, some of them may not get selected, but considering that this is a genetic algorithm I think that is fine. Just do this instead.

```
array indices, values;
sort(values, indices, samplez);
int n = indices.elements();
return (n > nsamples / 2) ? indices(seq(n - nsamples/2, n -1)) : indices;
```